<a href="https://colab.research.google.com/github/jgbedevi/test_technique_analyste/blob/master/notebooks/Untitled.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

  # **1. Imports**



In [92]:
import numpy as np
import pandas as pd




# **2.   Chargement des datasets**



In [93]:
df_communes = pd.read_csv("Documents/gyj/Test_technique_analyst/data/details_communes.csv")
df_centres = pd.read_csv("Documents/gyj/Test_technique_analyst/data/centres_service.csv")
df_socio = pd.read_csv("Documents/gyj/Test_technique_analyst/data/donnees_socioeconomiques.csv")
df_dev = pd.read_csv("Documents/gyj/Test_technique_analyst/data/developpement.csv")
df_routes = pd.read_csv("Documents/gyj/Test_technique_analyst/data/reseau_routier_togo_ext.csv")
df_docs = pd.read_csv("Documents/gyj/Test_technique_analyst/data/documents_administratifs_ext.csv")
df_demandes = pd.read_csv("Documents/gyj/Test_technique_analyst/data/demandes_service_public.csv")
df_logs = pd.read_csv("Documents/gyj/Test_technique_analyst/data/logs_activite.csv")


# **3. Aperçu structurel**


# **details_communes**

In [94]:
# Vérification des dimensions des dataframes
df_communes.shape

(200, 13)

In [95]:
# Vérification des types de variables
df_communes.dtypes

commune_id               object
commune                  object
prefecture               object
region                   object
latitude                float64
longitude               float64
altitude_m                int64
superficie_km2            int64
population_densite        int64
code_postal              object
type_commune             object
distance_capitale_km      int64
zone_climatique          object
dtype: object

In [96]:
# Aperçu du contenu
df_communes.head()


commune_id     commune prefecture    region  latitude  longitude  \
0     COM001       Vogan         Vo  Maritime    6.2295     1.5040   
1     COM002     Hahotoé         Vo  Maritime    6.6892     1.6588   
2     COM003  Anyronkopé         Vo  Maritime    6.4158     1.2398   
3     COM004        Momé         Vo  Maritime    6.2023     1.6180   
4     COM005     Dagbati         Vo  Maritime    6.3068     1.6507   

   altitude_m  superficie_km2  population_densite code_postal  type_commune  \
0          19             343                  35      BP 272        Rurale   
1          39             398                 210      BP 239  Semi-urbaine   
2          47             215                  76      BP 495        Rurale   
3          41             230                  95       BP 60        Rurale   
4          25             318                  58      BP 235        Rurale   

   distance_capitale_km zone_climatique  
0                    34         Côtière  
1                    79         Côtière  
2                    31         Côtière  
3                    45         Côtière  
4                    52         Côtière

**Aperçu dataset** : COMMUNES\
 **Dimensions :** (200 , 13)\
 **Types :** object + int    
 **Colonnes importantes :** commune_id, commune, prefecture, region, population, type_commune, distance_capitale, latitude, longitude\
 **Notes :** ce dataset répertorie les informations concernant les communes existant au Togo.

# **centres_service.csv**

In [97]:
# Vérification des dimensions des dataframes
df_centres.shape

(55, 16)

In [98]:
# Vérification des types de variables
df_centres.dtypes

centre_id                   object
nom_centre                  object
type_centre                 object
region                      object
prefecture                  object
commune                     object
quartier                    object
latitude                   float64
longitude                  float64
personnel_capacite_jour      int64
nombre_guichets              int64
heures_ouverture            object
horaire_nuit                object
equipement_numerique        object
date_ouverture              object
statut_centre               object
dtype: object

In [99]:
# Aperçu du contenu
df_centres.head()


centre_id                nom_centre type_centre    region prefecture  \
0     CT001  Centre Principal Dapaong   Principal   Savanes       Tône   
1     CT002               Centre Kévé  Secondaire  Maritime        Ave   
2     CT003              Centre Gando  Secondaire   Savanes    Oti-Sud   
3     CT004            Centre Pagouda  Secondaire      Kara      Binah   
4     CT005            Centre Tchamba  Secondaire  Centrale    Tchamba   

   commune           quartier  latitude  longitude  personnel_capacite_jour  \
0  Dapaong      Administratif   10.5920     0.4742                      300   
1     Kévé  Zone industrielle    6.2815     1.2182                       63   
2    Gando  Zone industrielle   10.2629     0.2525                       98   
3  Pagouda      Gare routière    9.4247     1.1698                       78   
4  Tchamba          Carrefour    8.5008     0.9988                       62   

   nombre_guichets heures_ouverture horaire_nuit equipement_numerique  \
0               12      07:30-17:30          Non              Complet   
1                2      08:00-16:00          Non               Limite   
2                2      08:00-16:00          Non              Partiel   
3                6      08:00-16:00          Non              Partiel   
4                3      08:00-16:00          Non               Limite   

  date_ouverture statut_centre  
0     2016-02-14         Actif  
1     2020-09-14         Actif  
2     2015-06-02         Actif  
3     2019-12-03         Actif  
4     2022-10-20         Actif

**Aperçu dataset :** \
**CENTRES Dimensions :** (55 , 16) \
**Types :** object + int + float \
**Colonnes importantes :** centre_id, nom_centre, type_centre, prefecture, region, commune, quartier, latitude,longitude, personnel_capacite_jour, nombre_guichets, equipement_numerique, statut_centre \
**Notes :** ce dataset répertorie les informations concernant les centres de traitement des demandes des documents officiel au Togo.

#**donnees_socioecomiques.csv**

In [100]:
# Vérification des dimensions des dataframes
df_socio.shape

(115, 11)

In [101]:
# Vérification des types de variables
df_socio.dtypes

region                   object
prefecture               object
commune                  object
population                int64
superficie_km2            int64
densite                   int64
taux_urbanisation       float64
taux_alphabétisation    float64
age_median                int64
nombre_menages            int64
revenu_moyen_fcfa         int64
dtype: object

In [102]:
# Aperçu du contenu
df_socio.head()


region prefecture     commune  population  superficie_km2  densite  \
0  Maritime         Vo       Vogan       48777             478      102   
1  Maritime         Vo     Hahotoé       13939             444       31   
2  Maritime         Vo  Anyronkopé       23395             355       65   
3  Maritime         Vo        Momé       19379             302       64   
4  Maritime         Vo     Dagbati       14846             309       48   

   taux_urbanisation  taux_alphabétisation  age_median  nombre_menages  \
0               0.33                  0.76          19            8590   
1               0.35                  0.66          21            3973   
2               0.18                  0.72          21            4534   
3               0.10                  0.81          22            4207   
4               0.32                  0.79          22            3170   

   revenu_moyen_fcfa  
0              78348  
1             112380  
2             107120  
3              99265  
4             133012

**Aperçu dataset :** Socio-economique \
**Dimensions :** (115 , 11) \
**Types :** object + int + float \
**Colonnes importantes :** region, prefecture, commune, population, superficie_km2, densite, taux_urbanisation, taux_alphabétisation, age_median, nombre_menages, revenu_moyen_fcfa \
**Notes** : ce dataset répertorie les informations socioéconomique de la population par région,préfecture et commune au Togo.

# **developpement.csv**

In [103]:
# Vérification des dimensions des dataframes
df_dev.shape

(33, 15)

In [104]:
# Vérification des types de variables
df_dev.dtypes

region                    object
prefecture                object
commune                   object
pib_par_habitant_fcfa      int64
taux_chomage             float64
taux_pauvrete            float64
acces_electricite        float64
acces_eau_potable        float64
acces_internet           float64
indice_developpement     float64
score_education          float64
score_sante              float64
nombre_ecoles              int64
nombre_hopitaux            int64
nombre_banques             int64
dtype: object

In [105]:
# Aperçu du contenu
df_dev.head()


region prefecture   commune  pib_par_habitant_fcfa  taux_chomage  \
0  Maritime         Vo      Lomé                 860000          0.11   
1  Maritime         Vo  Adakpamé                 830000          0.12   
2  Maritime      Golfe        Bè                 720000          0.13   
3  Maritime      Golfe      Agoè                 700000          0.14   
4  Maritime       Lacs     Aného                 560000          0.16   

   taux_pauvrete  acces_electricite  acces_eau_potable  acces_internet  \
0           0.24               0.96               0.93            0.67   
1           0.26               0.94               0.91            0.63   
2           0.27               0.92               0.89            0.60   
3           0.29               0.91               0.88            0.58   
4           0.31               0.86               0.83            0.46   

   indice_developpement  score_education  score_sante  nombre_ecoles  \
0                  0.74             0.79         0.76            250   
1                  0.71             0.77         0.74            220   
2                  0.69             0.75         0.73            190   
3                  0.67             0.73         0.71            175   
4                  0.63             0.69         0.66             98   

   nombre_hopitaux  nombre_banques  
0               19              48  
1               16              40  
2               13              30  
3               12              26  
4                6              16

**Aperçu dataset :** Developpement \
**Dimensions** : (33 , 15) \
**Types :** object + int + float
**Colonnes importantes :** region, prefecture, commune, pib_par_habitant_fcfa, taux_chomage, taux_pauvrete, acces_electricite, acces_eau_potable, acces_internet, indice_developpement, score_education, score_sante, nombre_ecoles, nombre_hopitaux, nombre_banques \
**Notes :** ce dataset contient les indicateurs de mesure du développement par region, prefecture, communel au Togo.

# **reseau_routier_togo_ext.csv**

In [106]:
# Vérification des dimensions des dataframes
df_routes.shape

(40, 14)

In [107]:
# Vérification des types de variables
df_routes.dtypes

route_id                   object
type_route                 object
nom_route                  object
region_origine             object
region_destination         object
prefecture_origine         object
prefecture_destination     object
longueur_km                 int64
etat_route                 object
temps_parcours_heures     float64
points_controle             int64
bus_par_jour                int64
camions_par_jour            int64
passagers_par_jour          int64
dtype: object

In [108]:
# Aperçu du contenu
df_routes.head()


route_id type_route nom_route region_origine region_destination  \
0     R001  Nationale       RN1       Maritime           Plateaux   
1     R002  Nationale       RN2       Plateaux           Centrale   
2     R003  Nationale       RN3       Centrale               Kara   
3     R004  Nationale       RN4           Kara            Savanes   
4     R005  Nationale       RN5       Maritime               Kara   

  prefecture_origine prefecture_destination  longueur_km etat_route  \
0                 Vo                   Ogou          150        Bon   
1               Ogou              Sotouboua          120      Moyen   
2          Sotouboua                  Kozah          110      Moyen   
3              Kozah                    Oti           95   Mediocre   
4                 Vo                  Kozah          420      Moyen   

   temps_parcours_heures  points_controle  bus_par_jour  camions_par_jour  \
0                    2.5                3            85                45   
1                    2.8                2            65                35   
2                    2.6                2            60                30   
3                    3.0                2            45                25   
4                    8.5                6            70                55   

   passagers_par_jour  
0                1250  
1                 950  
2                 850  
3                 650  
4                1600

**Aperçu dataset :** RESEAUX ROUTIERS \
**Dimensions :** (40 , 14) \
**Types :** object + int + float \
**Colonnes importantes :** route_id type_route nom_route region_origine region_destination prefecture_origine prefecture_destination longueur_km etat_route temps_parcours_heures points_controle bus_par_jour camions_par_jour passagers_par_jour \
**Notes :** ce dataset répertorie les informations concernant les réseaux routiers au Togo.

# **demandes_service_public.csv**

In [109]:
# Vérification des dimensions des dataframes
df_demandes.shape

(600, 16)

In [110]:
# Vérification des types de variables
df_demandes.dtypes

demande_id                 object
region                     object
prefecture                 object
commune                    object
quartier                   object
type_document              object
categorie_document         object
nombre_demandes             int64
delai_traitement_jours      int64
taux_rejet                float64
date_demande               object
motif_demande              object
statut_demande             object
canal_demande              object
age_demandeur               int64
sexe_demandeur             object
dtype: object

In [111]:
# Aperçu du contenu
df_demandes.head()


demande_id    region prefecture    commune     quartier  \
0       D001  Centrale  Sotouboua  Sotouboua  Nyékonakpoé   
1       D002      Kara      Binah    Pagouda      Hôpital   
2       D003  Plateaux       Ogou   Atakpamé           Bè   
3       D004   Savanes        Oti      Mango     Amoutivé   
4       D005  Maritime        Zio     Tsévié    Adidogomé   

               type_document categorie_document  nombre_demandes  \
0           Carte d'identité           Identité              168   
1           Carte d'identité           Identité               17   
2  Certificat de nationalité           Identité              104   
3           Carte d'identité           Identité               92   
4           Livre de famille              Civil              138   

   delai_traitement_jours  taux_rejet date_demande     motif_demande  \
0                       1        0.00   2023-09-30             Perte   
1                      22        0.09   2023-12-22               Vol   
2                       4        0.08   2023-08-13          Concours   
3                      39        0.02   2023-06-08  Première demande   
4                       7        0.12   2023-02-02     Détérioration   

  statut_demande canal_demande  age_demandeur sexe_demandeur  
0        Rejetée      En ligne             27              M  
1        Traitee      En ligne             21              F  
2        Traitee      En ligne             18              M  
3        Rejetée       Guichet             64              M  
4        Rejetée      En ligne             30              M

**Aperçu dataset :** DEMANDES SERVICE PUBLIC \
**Dimensions :** (600 , 16) \
**Types :** object + int + float \
**Colonnes importantes :** demande_id region prefecture commune quartier type_document categorie_document nombre_demandes delai_traitement_jours taux_rejet date_demande motif_demande statut_demande canal_demande age_demandeur sexe_demandeur \
**Notes :** ce dataset contient les informations sur chaque demande de documents officiel au Togo.

# **documents_admistratifs.csv**

In [112]:
# Vérification des dimensions des dataframes
df_docs.shape

(64, 9)

In [113]:
# Vérification des types de variables
df_docs.dtypes

annee                  int64
mois                   int64
region                object
prefecture            object
commune               object
type_document         object
nombre_demandes        int64
delai_moyen_jours    float64
taux_rejet_moyen     float64
dtype: object

In [114]:
# Aperçu du contenu
df_docs.head()


annee  mois    region prefecture commune      type_document  \
0   2023     1  Maritime         Vo    Lomé   Carte d'identité   
1   2023     1  Maritime         Vo    Lomé          Passeport   
2   2023     1  Maritime         Vo    Lomé  Acte de naissance   
3   2023     1  Maritime         Vo    Lomé   Livre de famille   
4   2023     1  Maritime       Lacs   Aného   Carte d'identité   

   nombre_demandes  delai_moyen_jours  taux_rejet_moyen  
0             2450               12.5              0.05  
1             1650               25.2              0.08  
2              890               10.1              0.02  
3              620               15.3              0.03  
4             1250               14.5              0.04

**Aperçu dataset :** DOCUMENTS ADMINISTRATIFS \
**Dimensions :** (64, 9) \
**Types :** object + int + float \
**Colonnes importantes :** annee mois region prefecture commune type_document nombre_demandes delai_moyen_jours taux_rejet_moyen \
**Notes :** ce dataset contient les informations agrégées sur les demandes par année, mois..., des documents officiel au Togo.

# **logs_activites.csv**

In [115]:
# Vérification des dimensions des dataframes
df_logs.shape

(450, 14)

In [116]:
# Vérification des types de variables
df_logs.dtypes

log_id                         object
centre_id                      object
date_operation                 object
type_operation                 object
type_document                  object
nombre_traite                   int64
delai_effectif                  int64
nombre_rejete                   int64
raison_rejet                   object
personnel_present               int64
temps_attente_moyen_minutes     int64
incident_technique             object
heure_debut                    object
heure_fin                      object
dtype: object

In [117]:
# Aperçu du contenu
df_logs.head()


log_id centre_id date_operation type_operation      type_document  \
0  LOG001     CT015     2023-10-30     Traitement   Carte d'identité   
1  LOG002     CT026     2023-04-21    Maintenance                NaN   
2  LOG003     CT049     2023-08-22     Traitement   Livre de famille   
3  LOG004     CT047     2023-07-13     Traitement  Acte de naissance   
4  LOG005     CT008     2023-04-08     Traitement          Passeport   

   nombre_traite  delai_effectif  nombre_rejete    raison_rejet  \
0            241              41             31    Non éligible   
1              0               0              0             NaN   
2             62              19              0             NaN   
3            139              22             12  Double demande   
4             71              35              6  Double demande   

   personnel_present  temps_attente_moyen_minutes incident_technique  \
0                 15                           58                Non   
1                 12                            0                Non   
2                 12                           95                Non   
3                  7                           54                Non   
4                 13                           94                Oui   

  heure_debut heure_fin  
0       08:00     16:00  
1       09:00     19:00  
2       08:30     17:30  
3       08:15     16:15  
4       08:30     17:30

**Aperçu dataset :** LOGS ACTIVITES \
**Dimensions :** (450 , 14) \
**Types :** object + int + float \
**Colonnes importantes :** log_id centre_id date_operation type_operation type_document nombre_traite delai_effectif nombre_rejete raison_rejet personnel_present temps_attente_moyen_minutes incident_technique heure_debut heure_fin \
**Notes** : ce dataset contient les informations de logs des traitements des demandes des documents officiel au Togo.

# **4. Identification des clés**


# **5. Notes et observations**
